sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveLateralColumnAliasReference.scala

```diff
@@ -170,7 +170,8 @@ object ResolveLateralColumnAliasReference extends Rule[LogicalPlan] {
           case (a: Alias, idx) =>
             val lcaResolved = unwrapLCAReference(a)
             // Insert the original alias instead of rewritten one to detect chained LCA
-            aliasMap += (a.toAttribute -> AliasEntry(a, idx))
+            aliasMap =
+              AttributeMap(aliasMap.baseMap.values.toMap + (a.toAttribute -> AliasEntry(a, idx)))
```

Comment:
```suggestion
              aliasMap ++= AttributeMap(Map(a.toAttribute -> AliasEntry(a, idx)))
```

Reply:
Maybe so that we can hide access to internal values as much as possible.

Reply:
Sounds good. But it seems created extra AttributeMap.
How about `aliasMap = AttributeMap(aliasMap.updated(a.toAttribute, AliasEntry(a, idx)))`?